lightning/src/ln/channelmanager.rs

```diff
@@ -680,12 +680,22 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref> ChannelManager<ChanSigner, M,
 	///
 	/// Raises APIError::APIMisuseError when channel_value_satoshis > 2**24 or push_msat is
 	/// greater than channel_value_satoshis * 1k or channel_value_satoshis is < 1000.
-	pub fn create_channel(&self, their_network_key: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_id: u64) -> Result<(), APIError> {
+	pub fn create_channel(&self, their_network_key: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_id: u64, override_config: Option<UserConfig>) -> Result<(), APIError> {
 		if channel_value_satoshis < 1000 {
 			return Err(APIError::APIMisuseError { err: "channel_value must be at least 1000 satoshis" });
 		}
 
-		let channel = Channel::new_outbound(&*self.fee_estimator, &self.keys_manager, their_network_key, channel_value_satoshis, push_msat, user_id, Arc::clone(&self.logger), &self.default_configuration)?;
+                let channel = match override_config {
```

Comment:
719b5ec

You may try
`
if override_config.is_some() { override_config.as_ref().unwrap() } else { &self.default_configuration }
`

Reply:
So in regards to the git jobs, I've squashed and dropped the fixup. My bad, looking back at the contribution guidelines I missed this: 

`Further, each commit, individually, should compile and pass tests, in order to ensure git bisect and other automated tools function properly.`

Reply:
thanks and appreciate the review @TheBlueMatt, I'll work on the request changes